src/DotNetWorker/OutputBindings/DefaultOutputBindingsInfoFactory.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Functions.Worker.OutputBindings
+{
+    internal class DefaultOutputBindingsInfoFactory : IOutputBindingsInfoFactory
+    {
+        public OutputBindingsInfo Build(FunctionMetadata metadata)
+        {
+            if (HasNoOutputBindings(metadata))
+            {
+                return NoOutputBindingsInfo.Instance;
+            }
+            else if (HasOnlyReturnBinding(metadata))
+            {
+                return new MethodReturnOutputBindingsInfo();
+            }
+            else
+            {
+                var bindingNames = metadata.OutputBindings.Keys.ToList();
+                return new PropertyOutputBindingsInfo(bindingNames);
+            }
+        }
+
+        private static bool HasOnlyReturnBinding(FunctionMetadata metadata)
+        {
+            if (metadata.OutputBindings.Any(kv => kv.Key.Equals(OutputBindingsConstants.ReturnBindingName, StringComparison.OrdinalIgnoreCase)))
+            {
+                int bindingCount = metadata.OutputBindings.Count;
+                if (bindingCount > 1)
+                {
+                    throw new InvalidOperationException($"You can only have 1 output binding if using '$return' output binding. " +
+                        $"Instead found {bindingCount} total bindings.");
+                }
+
```

Comment:
Is this information clear and actionable? Is the assignment always explicit? Do we need a link to a doc?

Reply:
A link to doc might help. I will use this link -- https://aka.ms/dotnet-worker-poco-binding